pkgs/data/misc/media-types/default.nix

```diff
@@ -0,0 +1,80 @@
+{ lib, stdenv, fetchFromGitLab, nixosTests }:
+
+stdenv.mkDerivation rec {
+  pname = "media-types";
+  version = "10.1.0";
+
+  src = fetchFromGitLab {
+    domain = "salsa.debian.org";
+    owner = "debian";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-vLQHjHrYShEBf4Py0vjpugbfY8HHnIIrmnxIDjHu5g8=";
+  };
+
+  outputs = [ "out" "doc" ];
+
+  installPhase = ''
+    install -D -m0444 mime.types $out/etc/mime.types
+    install -D -m0444 debian/bug-presubj $doc/share/doc/media-types/presubj
+    install -D -m0444 debian/changelog $doc/share/doc/media-types/changelog
+    install -D -m0444 debian/copyright $doc/share/doc/media-types/copyright
+
+    # Generate nginx mime-types
+    mkdir -p  $out/etc/nginx
+    echo "types {" > $out/etc/nginx/mime.types
+    cat mime.types | sed -e '/^#.*/d' -e '/^$/d' -e 's/.*/\ \ &;/' -ne '/[^[:space:]][[:space:]]\{1,\}[^[:space:]]/p' >> $out/etc/nginx/mime.types
+    echo "}" >> $out/etc/nginx/mime.types
```

Comment:
Is there no better way to generate this which is more robust and can't break that easily? Maybe we should check this with nginx in the build if possible and not only with a passthru vm test

Reply:
A similar method is used in the mailcap script.